### DISCUSSION TITLE
Build folder getting bigger and bigger

### DISCUSSION THREAD

**Discussion by victorVazquez** [vercel/next.js#26220] [Help]
Hey guys can you help me with an issue we face in my current project. This is the thing:

**Context**
We have a site using NextJS but adjusted to the client requirements, so is not by the book as nextjs recommends. In order to publish the site we need to create a build from the NextJS application (That contains a lot of pages) and push it to an S3 bucket.

We were creating the build, zipping it, pushing it to S3, and then deleting all the files related to that build (zip, .next folder, etc), but that process was taking up to 10 minutes. We then found out that if we don’t delete the files related to the build (Let’s call it if we have an initial build) the process will take less than half.

**Problem**
Now, after a few build creation operation we realized out .next folder was getting bigger and bigger to a point where a build was taking more that 2 GB (A normal build of the site is just a couple hundreds MB). Our theory is that NextJS is storing data from old builds as well as from the new build with each of those operations and we haven’t been able to figure out why it is doing that or if there’s a config value either from NextJS of webpack to avoid this behavior.
Any possible help would be greatly appreciated.

Thanks in advanced.

**Comment by timneutkens** [comment]
Cleaning the distDir was landed in Next.js 11, RFC: https://github.com/vercel/next.js/discussions/6009

**Reply by victorVazquez** [reply]
That's pretty cool, thanks @timneutkens. Although, the problem we face is in every build, files are duplicated inside `.build/cache` and `.build/static` folders.

We found in this [article](https://blog.devgenius.io/deployment-of-nextjs-application-using-github-action-to-azure-37318daedf91) that the cache folder: 

> "...which seems like a temporary cache for the NextJS compiler. The folder can grow really big because every build increases the folder size and should definitely be omitted from your deployed artifacts. So this part would no longer be an issue."

As for the second, our issue is that the chunks and the build manifest files are duplicated. In this case we'd like to know if there's a way from nextjs configuration to prevent this duplicates every time a build is performed.

**Reply by timneutkens** [reply]
distDir is cleaned in Next.js 11 so `.next/static` will be deleted at the start of the build. `.next/cache` is the webpack/babel etc caching to make subsequent builds faster.

**Reply by victorVazquez** [reply]
Cool, thx @timneutkens.
